bench/Benchmark.hs

```diff
@@ -0,0 +1,77 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+module Main where
+
+import Control.Lens ((&), (.~))
+import Control.Monad (replicateM)
+import Control.Monad.Except (runExceptT)
+import Control.Monad.State (evalStateT, execStateT)
+import Criterion.Main
+import GHC.Int (Int64)
+import Linear.V2 (V2 (V2))
+import Swarm.Game.CEK (initMachine)
+import Swarm.Game.Robot (Robot, mkRobot, systemRobot)
+import Swarm.Game.State (GameState, addRobot, initGameState)
+import Swarm.Game.Step (gameTick)
+import qualified Swarm.Language.Context as Context
+import Swarm.Language.Pipeline (ProcessedTerm)
+import Swarm.Language.Pipeline.QQ (tmQ)
+import Swarm.Language.Syntax (Term (TInt))
+import System.Random (randomRIO)
+
+treeProgram :: Integer -> Integer -> ProcessedTerm
+treeProgram rn1 rn2 =
+  [tmQ|
+  let repeat : int -> cmd () -> cmd () = \n.\c.
+    if (n == 0) {} {c ; repeat (n-1) c}
+  in {
+    repeat ($int:rn1 + 500) wait;
+    appear "|";
+    repeat ($int:rn2 + 500) wait;
+    place "tree";
```

Comment:
It looks like each tree program will take 1000 + rn1 + rn2 + 3 or so ticks to run to completion, right?  So it seems odd to me that we are only running for 1000 ticks, when none of the tree programs will be finished.  But maybe that's on purpose?

Reply:
Good point, I forgot there were two waits! I'll change the constant to 300, even if that's different from the real value in the tree entity, because the benchmark would take too long otherwise.